backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/dsl/impl/ElasticsearchIndexFieldTypeFactoryImpl.java

```diff
@@ -250,4 +251,9 @@ public Gson getUserFacingGson() {
 	public ElasticsearchDefaultFieldFormatProvider getDefaultFieldFormatProvider() {
 		return defaultFieldFormatProvider;
 	}
+
+	@Override
+	public BackendMappingHints hints() {
+		return BackendMappingHints.NONE;
```

Comment:
This is very weird. The actual hints should be retrieved from the mapping, like in `org.hibernate.search.engine.backend.mapping.spi.BackendMappingContext#hints`. You'll need to create a new `hints` methods though, because this is bootstrap and the `BackendMappingContext` does not even exist yet.

Maybe some new `BackendMapperContext` that you would implement in `PojoMapper` and whose instance would be passed by `PojoMapper` to a new argument of `org.hibernate.search.engine.common.impl.MappedIndexManagerFactoryImpl#createMappedIndexManager` and eventually `org.hibernate.search.backend.elasticsearch.impl.ElasticsearchBackendImpl#createIndexManagerBuilder`, to be stored in `org.hibernate.search.backend.elasticsearch.document.model.dsl.impl.ElasticsearchIndexRootBuilder` and eventually used in the type factory when it's created in `org.hibernate.search.backend.elasticsearch.document.model.dsl.impl.ElasticsearchIndexRootBuilder#createTypeFactory`?